AG-11264 - Use `ag-website-shared` components

taktran opened the pull request:
* Migrate icons, search box and pricing page to https://github.com/ag-grid/ag-website-shared

taktran committed:
git subrepo clone --branch=latest git@github.com:ag-grid/ag-website-shared.git packages/ag-charts-website/src/ag-website-shared

subrepo:
  subdir:   "packages/ag-charts-website/src/ag-website-shared"
  merged:   "7d10d53b6"
upstream:
  origin:   "git@github.com:ag-grid/ag-website-shared.git"
  branch:   "latest"
  commit:   "7d10d53b6"
git-subrepo:
  version:  "0.4.6"
  origin:   "https://github.com/ingydotnet/git-subrepo"
  commit:   "73a0129"

taktran committed:
AG-11264 - Add type alias for shared folder

taktran committed:
AG-11264 - Update icons to use shared repo on charts

taktran committed:
git subrepo pull packages/ag-charts-website/src/ag-website-shared

subrepo:
  subdir:   "packages/ag-charts-website/src/ag-website-shared"
  merged:   "7d9f0164d"
upstream:
  origin:   "git@github.com:ag-grid/ag-website-shared.git"
  branch:   "latest"
  commit:   "d0b9dc05f"
git-subrepo:
  version:  "0.4.6"
  origin:   "https://github.com/ingydotnet/git-subrepo"
  commit:   "73a0129"

taktran committed:
git subrepo push packages/ag-charts-website/src/ag-website-shared

subrepo:
  subdir:   "packages/ag-charts-website/src/ag-website-shared"
  merged:   "c678ad378"
upstream:
  origin:   "git@github.com:ag-grid/ag-website-shared.git"
  branch:   "latest"
  commit:   "c678ad378"
git-subrepo:
  version:  "0.4.6"
  origin:   "https://github.com/ingydotnet/git-subrepo"
  commit:   "73a0129"

taktran committed:
AG-11264 - Move license pricing to shared repo and make it configurable

taktran committed:
git subrepo push packages/ag-charts-website/src/ag-website-shared

subrepo:
  subdir:   "packages/ag-charts-website/src/ag-website-shared"
  merged:   "c8a55c691"
upstream:
  origin:   "git@github.com:ag-grid/ag-website-shared.git"
  branch:   "latest"
  commit:   "c8a55c691"
git-subrepo:
  version:  "0.4.6"
  origin:   "https://github.com/ingydotnet/git-subrepo"
  commit:   "73a0129"

taktran committed:
AG-11264 - Move framework images to shared folder

taktran committed:
AG-11264 - Use images from shared repo

taktran committed:
git subrepo push packages/ag-charts-website/src/ag-website-shared

subrepo:
  subdir:   "packages/ag-charts-website/src/ag-website-shared"
  merged:   "86476b061"
upstream:
  origin:   "git@github.com:ag-grid/ag-website-shared.git"
  branch:   "latest"
  commit:   "86476b061"
git-subrepo:
  version:  "0.4.6"
  origin:   "https://github.com/ingydotnet/git-subrepo"
  commit:   "73a0129"

taktran committed:
git subrepo pull packages/ag-charts-website/src/ag-website-shared

subrepo:
  subdir:   "packages/ag-charts-website/src/ag-website-shared"
  merged:   "8ad966d6d"
upstream:
  origin:   "git@github.com:ag-grid/ag-website-shared.git"
  branch:   "latest"
  commit:   "8ad966d6d"
git-subrepo:
  version:  "0.4.6"
  origin:   "https://github.com/ingydotnet/git-subrepo"
  commit:   "73a0129"

taktran committed:
AG-11264 - Fix formatting

taktran committed:
git subrepo push packages/ag-charts-website/src/ag-website-shared

subrepo:
  subdir:   "packages/ag-charts-website/src/ag-website-shared"
  merged:   "bde055706"
upstream:
  origin:   "git@github.com:ag-grid/ag-website-shared.git"
  branch:   "latest"
  commit:   "bde055706"
git-subrepo:
  version:  "0.4.6"
  origin:   "https://github.com/ingydotnet/git-subrepo"
  commit:   "73a0129"

alantreadway commented on packages/ag-charts-website/src/ag-website-shared/.gitrepo:
I'm not really a fan of the nesting here - can we skip the top level folders and just pull in `src/` to:
- avoid all of the unecessary and potentially confusing configuration files in the root
- avoid repeated `src` folder in the path 

Alternatively/additionally, could we maybe pull this in somewhere more obvious in the repo structure, e.g.:
`{workspareRoot}/external/ag-website-shared`?

taktran replied:
I can have a look at trying `{workspareRoot}/external/ag-website-shared` or similar. I'll need to see how imports work, or whether we'll need to do some extra npm/file linking. Hot reloading will also prob need an update, since it's outside of the website folder. The benefit is that it will only need to be done once in this setup phase.

> avoid all of the unecessary and potentially confusing configuration files in the root

I'm hoping no one apart from the website team needs to look in this folder, so it won't be as confusing, as they will know why it's there.

> avoid repeated src folder in the path

I tried to mitigate the problem by updating the import aliases in`tsconfig.json` to have `"@ag-website-shared/*": ["src/*"]`, so imports ignore the extra `src` path eg, `@ag-website-shared/components/icon/Icon`. The only place that requires the actual path is `markdoc.config.ts` ie, `component('./src/ag-website-shared/src/components/icon/EnterpriseIcon', 'EnterpriseIcon')`. Most consumers of the shared folder won't need to know about it.

taktran replied:
Turns out, it was quite easy to move to `{workspareRoot}/external/ag-website-shared`. Just moved the files and updated the references to point to it. Changing the `tsconfig.json` appears to give us hot reloading for free (vite/astro probably knows how to handle it).

I do agree, it's a better home for the files.

alantreadway replied:
Sounds great - thank you!

taktran committed:
AG-11264 - Move shared subrepo to external folder

taktran committed:
git subrepo pull external/ag-website-shared

subrepo:
  subdir:   "external/ag-website-shared"
  merged:   "22c372663"
upstream:
  origin:   "git@github.com:ag-grid/ag-website-shared.git"
  branch:   "latest"
  commit:   "22c372663"
git-subrepo:
  version:  "0.4.6"
  origin:   "https://github.com/ingydotnet/git-subrepo"
  commit:   "73a0129"

taktran committed:
AG-11264 - Move shared references to external folder

taktran committed:
git subrepo pull external/ag-website-shared

subrepo:
  subdir:   "external/ag-website-shared"
  merged:   "4f4402def"
upstream:
  origin:   "git@github.com:ag-grid/ag-website-shared.git"
  branch:   "latest"
  commit:   "4f4402def"
git-subrepo:
  version:  "0.4.6"
  origin:   "https://github.com/ingydotnet/git-subrepo"
  commit:   "73a0129"

taktran committed:
AG-11264 - Update file references in shared module scss

taktran committed:
git subrepo push external/ag-website-shared

subrepo:
  subdir:   "external/ag-website-shared"
  merged:   "902ca1029"
upstream:
  origin:   "git@github.com:ag-grid/ag-website-shared.git"
  branch:   "latest"
  commit:   "902ca1029"
git-subrepo:
  version:  "0.4.6"
  origin:   "https://github.com/ingydotnet/git-subrepo"
  commit:   "73a0129"

taktran committed:
AG-11264 - Move website design system to legacy folder

Use shared design system instead. Only have `legacy-design-system/modules` left, and will be removed at a later date.

Add `external` folder to allowed file list for vite

taktran committed:
AG-11264 - Remove css modules files that have already been moved to shared folder

taktran committed:
AG-11264 - Move search to use shared component

taktran committed:
AG-11264 - Update subrepo parent after a rebase

markdurrant committed:
git subrepo pull external/ag-website-shared

subrepo:
  subdir:   "external/ag-website-shared"
  merged:   "ca18493fa"
upstream:
  origin:   "git@github.com:ag-grid/ag-website-shared.git"
  branch:   "latest"
  commit:   "ca18493fa"
git-subrepo:
  version:  "0.4.6"
  origin:   "git@github.com:ingydotnet/git-subrepo.git"
  commit:   "73a0129"

markdurrant committed:
git subrepo pull external/ag-website-shared

subrepo:
  subdir:   "external/ag-website-shared"
  merged:   "a525f158e"
upstream:
  origin:   "git@github.com:ag-grid/ag-website-shared.git"
  branch:   "latest"
  commit:   "a525f158e"
git-subrepo:
  version:  "0.4.6"
  origin:   "git@github.com:ingydotnet/git-subrepo.git"
  commit:   "73a0129"

kylerphillips committed:
git subrepo pull external/ag-website-shared

subrepo:
  subdir:   "external/ag-website-shared"
  merged:   "cf2763f4a"
upstream:
  origin:   "git@github.com:ag-grid/ag-website-shared.git"
  branch:   "latest"
  commit:   "cf2763f4a"
git-subrepo:
  version:  "0.4.6"
  origin:   "https://github.com/ingydotnet/git-subrepo"
  commit:   "73a0129"

kylerphillips committed:
AG-11264 Replace product dropdown reference

markdurrant committed:
git subrepo pull external/ag-website-shared

subrepo:
  subdir:   "external/ag-website-shared"
  merged:   "ab12a9e10"
upstream:
  origin:   "git@github.com:ag-grid/ag-website-shared.git"
  branch:   "latest"
  commit:   "ab12a9e10"
git-subrepo:
  version:  "0.4.6"
  origin:   "git@github.com:ingydotnet/git-subrepo.git"
  commit:   "73a0129"

markdurrant committed:
AG-11264 - Move <Alert/> component to website shared

markdurrant committed:
AG-11264 - Fix typo

markdurrant committed:
git subrepo push external/ag-website-shared

subrepo:
  subdir:   "external/ag-website-shared"
  merged:   "51f65db70"
upstream:
  origin:   "git@github.com:ag-grid/ag-website-shared.git"
  branch:   "latest"
  commit:   "51f65db70"
git-subrepo:
  version:  "0.4.6"
  origin:   "git@github.com:ingydotnet/git-subrepo.git"
  commit:   "73a0129"

taktran committed:
AG-11264 - Update parent sha of .gitrepo
